### ISSUE TITLE
Fix rendering of literals - Postgres module

### ISSUE THREAD

**Issue by robmwalsh** [zio/zio-sql#311]
Rendering of literals is broken for all modules - it's currently just using `.toString` on the underlying values. Using the type tag in the Expr.Literal, we need to determine the type of the enclosed `value` and render it using the provided builder. Bonus points for escaping to prevent SQL injection, but there's another ticket (#219) for that so just something that works and assumes benevolent users will do for now.

I've just opened one ticket for postgres - most of this work should be able to be copied between modules and adjusted rather than reinventing the wheel. I'll create tickets for the other modules once this one is complete.

```scala
override def renderRead(read: self.Read[_]): String = {
    ...
    def buildExpr[A, B](expr: self.Expr[_, A, B]): Unit = expr match {
      case Expr.Source(tableName, column)                               =>
     ...
      case Expr.Literal(value)                                          =>
           val _ = builder.append(value.toString) //<------this doesn't work real well D:
    }
  ...
}

**Comment by robmwalsh** [comment]
I'll take this
